appshell/appshell_extensions_platform.cpp

```diff
@@ -0,0 +1,31 @@
+#include "appshell/appshell_extensions_platform.h"
+#include <unicode/ucsdet.h>
+
+void GetCharsetMatch(const char* bufferData, size_t bufferLength, std::string &detectedCharSet) {
+	detectedCharSet = "";
+	const UCharsetMatch* charsetMatch_;
+	UErrorCode icuError = U_ZERO_ERROR;
+
+	UCharsetDetector* charsetDetector_ = ucsdet_open(&icuError);
+	if (U_FAILURE(icuError))
+		throw "Failed to open detector";
+
+	// send text
+	ucsdet_setText(charsetDetector_, bufferData, bufferLength, &icuError);
+	if (U_FAILURE(icuError))
```

Comment:
`ucsdet_close()` will go uncalled leading to a dirty state. Need to implement proper cleanups.